src/chunk.c

```diff
@@ -1790,53 +1977,195 @@ ts_chunk_get_by_relid(Oid relid, int16 num_constraints, bool fail_if_not_found)
+
+	return count == 1;
+}
+
+static bool
+chunk_simple_scan_by_name(const char *schema, const char *table, FormData_chunk *form,
+						  bool missing_ok)
+{
+	ScanIterator iterator;
+
+	if (schema == NULL || table == NULL)
+		return false;
+
+	iterator = ts_scan_iterator_create(CHUNK, AccessShareLock, CurrentMemoryContext);
+	init_scan_by_name(&iterator, schema, table);
+
+	return chunk_simple_scan(&iterator, form, missing_ok);
+}
+
+static bool
+chunk_simple_scan_by_relid(Oid relid, FormData_chunk *form, bool missing_ok)
+{
+	bool found = false;
+
+	if (OidIsValid(relid))
+	{
+		const char *table = get_rel_name(relid);
+
+		if (table != NULL)
+		{
+			Oid nspid = get_rel_namespace(relid);
+			const char *schema = get_namespace_name(nspid);
+
+			found = chunk_simple_scan_by_name(schema, table, form, missing_ok);
+		}
+	}
+
+	if (!found && !missing_ok)
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_OBJECT),
+				 errmsg("chunk with relid %u not found", relid)));
+
+	return found;
+}
+
+static bool
+chunk_simple_scan_by_id(int32 chunk_id, FormData_chunk *form, bool missing_ok)
+{
+	ScanIterator iterator;
+
+	iterator = ts_scan_iterator_create(CHUNK, AccessShareLock, CurrentMemoryContext);
+	init_scan_by_chunk_id(&iterator, chunk_id);
+
+	return chunk_simple_scan(&iterator, form, missing_ok);
+}
+
+/*
+ * Lookup a Chunk ID from a chunk's relid.
  */
 Datum
 ts_chunk_id_from_relid(PG_FUNCTION_ARGS)
 {
 	static Oid last_relid = InvalidOid;
 	static int32 last_id = 0;
 	Oid relid = PG_GETARG_OID(0);
-	Chunk *chunk;
+	FormData_chunk form;
 
 	if (last_relid == relid)
 		return last_id;
 
-	chunk = ts_chunk_get_by_relid(relid, 0, true);
+	chunk_simple_scan_by_relid(relid, &form, false);
+
 	last_relid = relid;
-	last_id = chunk->fd.id;
-	return last_id;
+	last_id = form.id;
+
+	PG_RETURN_INT32(last_id);
 }
 
-Chunk *
-ts_chunk_get_by_id(int32 id, int16 num_constraints, bool fail_if_not_found)
+bool
+ts_chunk_exists_relid(Oid relid)
 {
-	ScanKeyData scankey[1];
+	FormData_chunk form;
 
-	/*
-	 * Perform an index scan on chunk id.
-	 */
-	ScanKeyInit(&scankey[0], Anum_chunk_idx_id, BTEqualStrategyNumber, F_INT4EQ, Int32GetDatum(id));
+	return chunk_simple_scan_by_relid(relid, &form, true);
+}
+
+/*
+ * Get the relid of a chunk given its ID.
+ */
+Oid
+ts_chunk_get_relid(int32 chunk_id, bool missing_ok)
+{
+	FormData_chunk form = { 0 };
+	Oid relid = InvalidOid;
 
-	return chunk_scan_find(CHUNK_ID_INDEX,
-						   scankey,
-						   1,
-						   num_constraints,
-						   CurrentMemoryContext,
-						   fail_if_not_found);
+	if (chunk_simple_scan_by_id(chunk_id, &form, missing_ok))
+	{
+		Oid schemaid = get_namespace_oid(NameStr(form.schema_name), missing_ok);
+
+		if (OidIsValid(schemaid))
+			relid = get_relname_relid(NameStr(form.table_name), schemaid);
+	}
+
+	if (!OidIsValid(relid) && !missing_ok)
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_SCHEMA),
+				 errmsg("chunk with relid %u not found", relid)));
```

Comment:
Why is the error given in terms of `relid`, not `chunk_id? Should it be:
```suggestion
				 errmsg("chunk with id %u not found", chunk_id)));
```